DynamicAnalyzer/views/android/environment.py

```diff
@@ -40,41 +40,49 @@ def wait(self, sec):
         logger.info('Waiting for %s seconds...', str(sec))
         time.sleep(sec)
 
+    def proof_connect(self, output):
+        """Proof if connection output is not containing errors"""
+        if b'unable to connect' in output or b'failed to connect' in output:
+            logger.error('%s', output.decode('utf-8').replace('\n', ''))
+            return False
+        else:
+            return True
+
+    def run_subprocess_verify_output(self, command):
+        out = subprocess.check_output(command)
+        self.wait(2)
+        if not proof_connect(out):
+            return False
+        else:
+            return True
+
     def connect_n_mount(self):
         """Test ADB Connection."""
         self.adb_command(['kill-server'])
         self.adb_command(['start-server'])
         logger.info('ADB Restarted')
         self.wait(2)
         logger.info('Connecting to Android %s', self.identifier)
-        out = subprocess.check_output([get_adb(), 'connect', self.identifier])
-        self.wait(2)
-        if b'unable to connect' in out or b'failed to connect' in out:
-            logger.error('%s', out.decode('utf-8').replace('\n', ''))
+        if not run_subprocess_verify_output([get_adb(),
+                                            'connect',
+                                             self.identifier]):
             return False
```

Comment:
`return self. run_subprocess_verify_output..`

Reply:
Then the code will stop after that, isn't it? So I just want to only return False if it is False, otherwise continue and return True at the end.

Reply:
Yes you are right.